pkg/file/pipeline/pipeline.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pipeline
+
+import (
+	"github.com/ethersphere/bee/pkg/storage"
+)
+
+type pipeWriteArgs struct {
```

Comment:
This structure looks very similar to what the Chunk is. Could it be removed?

Reply:
well, no. since i need a way to pass references and unencrypted spans to the trie writer when doing encrypted content. https://github.com/ethersphere/bee/pull/613/files#diff-2fa7bcdc8b2ddbe0c7c7d1b4fb852b23R18

Reply:
That is not part of this PR, not sure what problems would it make with encryption, but that this type defined in abstraction does create a coupling with a specific (encrypt) writer in that sense.

Reply:
we will have more complexities and dependencies created between different writers in the pipeline when erasure coding and entanglement codes are introduced. having this struct makes _some_ things easier to express declaratively in the code. If we remove this type then I'd just have to offload all of these members into the writer method signature. There will be coupling here either way

Reply:
Yes, I see, the pipeline needs to be coupled because of complexities of different writers. This is what makes my most significant concern, which is having everything in one package, making abstraction leaks very easy.

Reply:
Fair enough. I see and respect your concerns. It is only a question to which degree we need abstraction within this component of pipeline. Food for thought in the meanwhile.